Core/Services/AuthenticationService.cs

```diff
@@ -110,23 +107,25 @@ private async Task<RefreshToken> CreateRefreshToken(string userId)
                 UserId = userId
             };
 
-            await _refreshTokenRepository.InsertAsync(refreshTokenEntity);
-            await _refreshTokenRepository.SaveChangesAsync();
+            await _refreshTokenRepository.AddAsync(refreshTokenEntity);
 
             return refreshTokenEntity;
         }
 
         public async Task<UserAutorizationDTO> RefreshTokenAsync(UserAutorizationDTO userTokensDTO)
         {
-            var refreshToken = GetRefreshToken(userTokensDTO.RefreshToken);
+            var refreshToken = await _refreshTokenRepository
+                .GetBySpecAsync(new GetRefreshTokenByToken(userTokensDTO.RefreshToken));
+
+            ExceptionMethods.RefreshTokenNullCheck(refreshToken);
```

Comment:
Is it a good way to check information and make an exception handling ?

Reply:
We will make this in another pr